syntax = "proto3";

package RobotArm;

import "robot_arm.proto";

  //The rpc method definitions are defined with regards to the server
  //So SendUpdate will receive a ConfigRequest from the client
  //Given this syntax, it seems as though the ConfigRequest is "passed"
  //into the SendUpdate - but it's just syntax.
  //On the client end, the client will make the request, and hold onto
  //the "stub" of that request.

service ConfigurationUpdate {

  rpc SendUpdate (currentStatus) returns (robotConfiguration) {}

  rpc Move5 (moveRequest5) returns (currentStatus) {}

  rpc Move7 (moveRequest7) returns (currentStatus) {}

}
